### PR TITLE
fix(shwap/bitswap): update boxo and config

### PR DESCRIPTION
**Pull request by Wondertan**
Yet another change to the bitswap configuration
* Adds `MinTimeout`. 
  * Robusta was having hard times with tail latency being too long, skewing timeout estimations, and causing cascade of simulated DONT_Haves that aren't real.
  * Basically only needed on low-latency envs like robusta.
* Reverts wantlist limit for peers to 4096
  * We observed limit overflow with FN syncing over bitswap. The overflow wasn't reached in any subsequent runs after the limit was raised back

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -362,4 +362,4 @@ replace (
 	github.com/tendermint/tendermint => github.com/celestiaorg/celestia-core v1.43.0-tm-v0.34.35
 )
 
-replace github.com/ipfs/boxo => github.com/celestiaorg/boxo v0.0.0-20241114103253-16dd306892a7
+replace github.com/ipfs/boxo => github.com/celestiaorg/boxo v0.0.0-20241118020536-b0a7e7e3f8b6
```
Comment (cristaloleg) [inline go.mod:365]: What is inside this update?

Comment (Wondertan) [reply go.mod:365]: Yes

Comment (Wondertan) [reply go.mod:365]: Btw kudos to @waldiss for the idea

**Review by cristaloleg — APPROVED** [review]

File: go.mod (hunk 1 above)
Comment (walldiss) [reply go.mod:365]: MinTimeout support in bitswap

Comment (cristaloleg) [reply go.mod:365]: @Wondertan wrong `walldiss` 🤣

**Pull request merged**